boa/src/syntax/lexer/cursor.rs

```diff
@@ -15,25 +15,24 @@ impl<R> Cursor<R> {
     pub(super) fn pos(&self) -> Position {
         self.pos
     }
-
-    /// Advances the position to the next line.
+    /// Advances the position to the next column.
     #[inline]
-    fn next_column(&mut self) {
+    pub(super) fn next_column(&mut self) {
         let current_line = self.pos.line_number();
         let next_column = self.pos.column_number() + 1;
         self.pos = Position::new(current_line, next_column);
     }
 
-    /// Advances the position to the next column.
+    /// Advances the position to the next line.
     #[inline]
-    fn next_line(&mut self) {
+    pub(super) fn next_line(&mut self) {
         let next_line = self.pos.line_number() + 1;
         self.pos = Position::new(next_line, 1);
     }
 
     /// Performs a carriage return to modify the position in the source.
     #[inline]
-    fn carriage_return(&mut self) {
+    pub(super) fn carriage_return(&mut self) {
```

Comment:
I don't think this should be called from `super`. This should only be called from the cursor, when encountering a `\r` character.

Reply:
Fixed